extension/src/frequency.rs

```diff
@@ -430,6 +536,56 @@ pub fn topn_agg_with_skew_trans(
         {SpaceSavingTransState::topn_agg_from_type_id(skew, n as u32, typ, collation)}).internal()
 }
 
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+pub fn topn_agg_with_skew_bigint_trans(
+    state: Internal,
+    n: i32,
+    skew: f64,
+    value: Option<i64>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal> {
+    if n <= 0 {
+        pgx::error!("topn aggregate requires an n value > 0")
+    }
+    if skew <= 1.0 {
+        pgx::error!("topn aggregate requires a skew factor > 1.0")
+    }
+
+    let value = match value {
+        None => None,
+        Some(val) => unsafe {AnyElement::from_datum(val as pg_sys::Datum, false, pg_sys::INT8OID)}
+    };
+
+    space_saving_trans(unsafe{ state.to_inner() }, value, fcinfo, |typ, collation|
+        {SpaceSavingTransState::topn_agg_from_type_id(skew, n as u32, typ, collation)}).internal()
+}
+
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+pub fn topn_agg_with_skew_text_trans(
+    state: Internal,
+    n: i32,
+    skew: f64,
+    value: Option<&str>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal> {
+    if n <= 0 {
+        pgx::error!("topn aggregate requires an n value > 0")
+    }
+    if skew <= 1.0 {
+        pgx::error!("topn aggregate requires a skew factor > 1.0")
+    }
+
+    // Important: this variable will hold the lifetime of the varlena until we copy it in freq_agg_trans()
+    let text = value.map(pgx::rust_str_to_text_p);
```

Comment:
if we're doing this I'd recommend dust using an `Option<crate::raw::text>` and using `pg_detoast_datum_copy()`